vita3k/renderer/src/vulkan/scene.cpp

```diff
@@ -99,6 +99,18 @@ void update_sync_signal(SceGxmSyncObject *sync) {
     }
 }
 
+#ifdef __APPLE__
+void restride_stream(GXMStreamInfo &stream, uint32_t old_stride) {
+    const uint32_t new_stride = (old_stride + 3) & -4;
+    stream.size = (stream.size / old_stride) * new_stride;
```

Comment:
I believe it should be
```suggestion
    stream.size = ((stream.size + old_stride- 1) / old_stride) * new_stride;
```

Moreover, I believe it will be cleaner to compute `nb_vertex_input =  ((stream.size + old_stride- 1) / old_stride)` and then use it for the rest of this function.

Reply:
I think original code is correct.

As far as my understand, data structrue looks like below.
```
Original
┌────────────┬────────────┐
│    Data    │    Data    │
└────────────┴────────────┘
├───Stride───┼───Stride───┤
├───────stream.size───────┤

-------------------------------------------

After restriding
┌────────────────────┬────────────────────┐
│    Data    │Padding│    Data    │Padding│
└────────────────────┴────────────────────┘
├───────Stride───────┼───────Stride───────┤
├───────────────stream.size───────────────┤
```
So dividing total data size with old_stride will give number of data counts.
And multipling aligned stride to data counts will create new data array size.

Reply:
The stream.size value is computed here https://github.com/Vita3K/Vita3K/blob/cc7f2847dcb3a7878d0b04c75b04345bea03f2cb/vita3k/modules/SceGxm/SceGxm.cpp#L2093 , as you can see this value is tight, meaning that if there is some already existing padding at the end of the data, the padding bytes at the end of the last data won't be counted in. So if there is some padding at the end of the data (before adding this new padding to make the stride a multiple of 4), the last vertex index will be missing without rounding up instead of down.